lightning/impl/sgd_fast.pyx

```diff
@@ -53,6 +53,21 @@ cdef class ModifiedHuber(LossFunction):
         else:
             return 4.0 * y
 
+cdef class WeightedLoss(LossFunction):
+
+    cdef double[:] sample_weights
+    cdef LossFunction loss_func
+
+    def __init__(self, np.ndarray[double, ndim=1] sample_weights, LossFunction loss_func):
+        self.sample_weights = sample_weights
+        self.loss_func = loss_func
+
+    cpdef double loss(self, double p, double y, int i):
```

Comment:
An alternative would be `loss(self, double p, double y, double sw)`. This would require passing sample_weights to each algorithm but the method signature is cleaner. I liked the fact that the loss functions are independent of the dataset.